packages/AttributeAwarePhpDoc/src/AttributeAwareNodeFactoryCollector.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\AttributeAwarePhpDoc;
+
+use Rector\AttributeAwarePhpDoc\Contract\AttributeNodeAwareFactory\AttributeNodeAwareFactoryInterface;
+
+final class AttributeAwareNodeFactoryCollector
+{
+    /**
+     * @var AttributeNodeAwareFactoryInterface[]
+     */
+    private $attributeAwareNodeFactories = [];
+
+    /**
+     * @param AttributeNodeAwareFactoryInterface[] $attributeAwareNodeFactories
+     */
+    public function __construct(array $attributeAwareNodeFactories)
+    {
+        $this->attributeAwareNodeFactories = $attributeAwareNodeFactories;
+    }
+
+    /**
+     * @return AttributeNodeAwareFactoryInterface[]
+     */
+    public function provide(): array
+    {
+        return $this->attributeAwareNodeFactories;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getSupportedNodeClasses(): array
+    {
+        $supportedNodeClasses = [];
+        foreach ($this->attributeAwareNodeFactories as $attributeAwareNodeFactory) {
+            $supportedNodeClasses[] = $attributeAwareNodeFactory->getOriginalNodeClass();
+        }
+
+        return $supportedNodeClasses;
```

Comment:
What about using `array_map` here ? Something like : 

```php
<?php
return array_map(
    static function (AttributeAwareNodeFactory $factory): string {
        return $factory->getOriginalNodeClass();
    },
    $this->attributeAwareNodeFactories
);
```

Reply:
I find it very hard to read

Reply:
Also array walk, filter etc. have gifferent syntax, behavior, some reference, some return array. Often leads to bugs along with unclear code

Reply:
:+1: (we use a custom library of functional methods in our project to circumvent that.. I know Laravel offers that too)